### DISCUSSION TITLE
SubmitShipment BOL

### DISCUSSION THREAD

**Discussion by adam12104** [amzn/selling-partner-api-models#3162] [General]
Is the "BOL" (Bill Of Lading) number pre populated by Amazon or is this something that we can generate at the point of an API call? For example, I am currently submitting an ASN manually using the spreadsheet option via the website. The ASN number (Which I presume is the BOL number?) Is auto generated at the point of creation. This is the same for the pallet Identifier/s. Also if I am submitting a shipment confirmation via the API is an ASN still necessary?

Sorry for the "Idiotic" questions but it would be really helpful to know.

Thanks :).

**Comment by jrusse11** [comment]
Not yet using API's in my workflow-- but I know when creating an ARN manually through vendor central the reference number entered becomes the BOL#. We use our invoice# as the reference/BOL# to link the two for fighting the inevitable shortage claims that pop up when they are receiving.  

Not sure if that info is useful to you but hope it helps!

**Reply by adam12104** [reply]
Yes that does help, I interpret that as we can set the BOL to what we want at the point of ASN creation. I am now building a program that generates an EDIFACT file for the ASN and taking that route because I don't believe the API fully supports the ASN submission yet.

**Comment by MeenaAmz** [comment]
Please refer to details on BOL [here](https://developer-docs.amazon.com/sp-api/docs/vendor-retail-procurement-apis-v1-use-case-guide#bill-of-lading-bol). Also now Selling partner API does support [Submit Shipment confirmation](https://developer-docs.amazon.com/sp-api/docs/vendor-shipments-api-v1-reference#submitshipmentconfirmations).
